src/Icicle/Avalanche/Statement/Simp.hs

```diff
@@ -391,10 +430,16 @@ substXinS a_fresh name payload statements
 --      statements that do not have any external effect are silly
 --  * Constant folding for ifs
 --
-thresherNoAlpha :: (Hashable n, Eq n, Ord a) => a -> Statement a n Prim -> FixT (Fresh n) (Statement a n Prim)
+thresherNoAlpha
+ :: (Hashable n, Eq n, Ord a)
+ => a
+ -> Statement (Ann a n) n Prim
+ -> FixT (Fresh n) (Statement (Ann a n) n Prim)
 thresherNoAlpha a_fresh statements
  = transformUDStmt trans Map.empty statements
  where
+  xVar n
+   = XVar (a_fresh, Set.singleton n) n
   trans env s
    -- Check if it actually does anything:
    -- updates accumulators, returns a value, etc.
```

Comment:
yeah, that sounds fair. I'll put in some SCC to see how much `hasEffect` has an effect (heh) on the performance